torch/csrc/jit/codegen/cuda/test/test_gpu.cpp

```diff
@@ -22305,7 +22305,9 @@ TEST_F(NVFuserTest, FusionRAWSyncInsertionPlace4_CUDA) {
       // Record number of unary ops that modifies shared memory.
       if (uop->out()->isA<kir::TensorIndex>() &&
           uop->out()->as<kir::TensorIndex>()->view()->getMemoryType() ==
-              MemoryType::Shared) {
+              MemoryType::Shared &&
+          // Filter out initialization expressions
+          uop->in()->isA<kir::TensorIndex>()) {
```

Comment:
Is this an independent bug fix?

Reply:
This is not completely independent as this PR made it fail. We were not inserting any shared mem initializations up until this PR, which the pattern matcher in this test case need to take care of.